pkg/webservice/dao/scheduler-health.go

```diff
@@ -0,0 +1,37 @@
+/*
+ Licensed to the Apache Software Foundation (ASF) under one
+ or more contributor license agreements.  See the NOTICE file
+ distributed with this work for additional information
+ regarding copyright ownership.  The ASF licenses this file
+ to you under the Apache License, Version 2.0 (the
+ "License"); you may not use this file except in compliance
+ with the License.  You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+package dao
+
+type SchedulerHealthDAOInfo struct {
+	Healthy              bool                         `json:"healthy"`
+	SchedulingErrors     int                          `json:"schedulingErrors"`
+	FailedNodes          int                          `json:"failedNodes"`
+	UnhealthyPartition   map[string]*PartitionDAOInfo `json:"unhealthyPartitions"`
+	UnhealthyNodes       map[string]*NodeDAOInfo      `json:"unhealthyNodes"`
+	ReservationNodeRatio float32                      `json:"reservationRatio"`
```

Comment:
```
type SchedulerHealthDAOInfo struct {
   Healthy bool
   HealthChecks []HealthCheckInfo
}

type HealthCheckInfo struct {
   Name string
   Description string // describe what we are checking here
   Succeed bool
   DiagnosisMessage string // non-empty if succeed=false, some human-readable message to explain why this fails
}
```

Reply:
I think it makes no sense to collect the successful checks as well. I would remove the `Succeed` flag from the `HealthCheckInfo` and collect only the failed infos.

Reply:
It makes some sense to display what kind of checks have been included. When that gets rendered in UI, it could give us a list of checks, and the individual states. If this is not adding too much work, I would rather include this info.

Reply:
Ok. I have included it